From c1769f1a4578cacef3741f6d16788ec5c92d115c Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Wed, 29 May 2019 16:58:09 +0300
Subject: [PATCH 7/7] arch64: dts: fsl-imx8mq: Update DCSS, LCDIF and DSI
 clocks

Update the clocks for the above components, for better clock
calculation.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
---
 .../dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dtsi |  5 ++---
 .../dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi | 26 ++++------------------
 .../freescale/fsl-imx8mq-evk-lcdif-rm67191.dtsi    | 22 +-----------------
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi      |  8 +++----
 4 files changed, 11 insertions(+), 50 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dtsi
index 77a612b..903ac6b 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dtsi
@@ -30,16 +30,15 @@
 		 <&clk IMX8MQ_CLK_DISP_AXI_ROOT>,
 		 <&clk IMX8MQ_CLK_DISP_RTRM_ROOT>,
 		 <&clk IMX8MQ_CLK_DC_PIXEL>,
-		 <&clk IMX8MQ_CLK_DUMMY>,
 		 <&clk IMX8MQ_CLK_DISP_DTRC>;
-	clock-names = "apb", "axi", "rtrm", "pix_div", "pix_out", "dtrc";
+	clock-names = "apb", "axi", "rtrm", "pix", "dtrc";
 	assigned-clocks = <&clk IMX8MQ_CLK_DC_PIXEL>,
 			  <&clk IMX8MQ_CLK_DISP_AXI>,
 			  <&clk IMX8MQ_CLK_DISP_RTRM>;
 	assigned-clock-parents = <&clk IMX8MQ_VIDEO_PLL1_OUT>,
 				 <&clk IMX8MQ_SYS1_PLL_800M>,
 				 <&clk IMX8MQ_SYS1_PLL_800M>;
-	assigned-clock-rates = <594000000>,
+	assigned-clock-rates = <600000000>,
 			       <800000000>,
 			       <400000000>;
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi
index d801dba..bd5e447 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi
@@ -30,24 +30,17 @@
 		 <&clk IMX8MQ_CLK_DISP_AXI_ROOT>,
 		 <&clk IMX8MQ_CLK_DISP_RTRM_ROOT>,
 		 <&clk IMX8MQ_CLK_DC_PIXEL>,
-		 <&clk IMX8MQ_CLK_DUMMY>,
 		 <&clk IMX8MQ_CLK_DISP_DTRC>;
-	clock-names = "apb", "axi", "rtrm", "pix_div", "pix_out", "dtrc";
-
+	clock-names = "apb", "axi", "rtrm", "pix", "dtrc";
 	assigned-clocks = <&clk IMX8MQ_CLK_DC_PIXEL>,
 			  <&clk IMX8MQ_CLK_DISP_AXI>,
-			  <&clk IMX8MQ_CLK_DISP_RTRM>,
-			  <&clk IMX8MQ_VIDEO_PLL1_REF_SEL>,
-			  <&clk IMX8MQ_VIDEO_PLL1>;
+			  <&clk IMX8MQ_CLK_DISP_RTRM>;
 	assigned-clock-parents = <&clk IMX8MQ_VIDEO_PLL1_OUT>,
 				 <&clk IMX8MQ_SYS1_PLL_800M>,
-				 <&clk IMX8MQ_SYS1_PLL_800M>,
-				 <&clk IMX8MQ_CLK_25M>;
+				 <&clk IMX8MQ_SYS1_PLL_800M>;
 	assigned-clock-rates = <600000000>,
 			       <800000000>,
-			       <0>,
-			       <400000000>,
-			       <599999999>;
+			       <400000000>;
 
 	dcss_disp0: port@0 {
 		reg = <0>;
@@ -64,17 +57,6 @@
 
 &mipi_dsi {
 	status = "okay";
-	assigned-clocks = <&clk IMX8MQ_CLK_DSI_PHY_REF>,
-			  <&clk IMX8MQ_CLK_DSI_CORE>,
-			  <&clk IMX8MQ_VIDEO_PLL1_REF_SEL>,
-			  <&clk IMX8MQ_VIDEO_PLL1>;
-	assigned-clock-parents = <&clk IMX8MQ_VIDEO_PLL1_OUT>,
-				 <&clk IMX8MQ_SYS1_PLL_266M>,
-				 <&clk IMX8MQ_CLK_25M>;
-	assigned-clock-rates = <24000000>,
-			       <266000000>,
-			       <0>,
-			       <599999999>;
 
 	port@1 {
 		mipi_dsi_in: endpoint {
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-rm67191.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-rm67191.dtsi
index 325ed26..0d29bc5 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-rm67191.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-rm67191.dtsi
@@ -38,17 +38,8 @@
 	status = "okay";
 	max-res = <1080>, <1920>;
 
-	assigned-clocks = <&clk IMX8MQ_CLK_LCDIF_PIXEL>,
-			  <&clk IMX8MQ_VIDEO_PLL1_REF_SEL>,
-			  <&clk IMX8MQ_VIDEO_PLL1>;
-	assigned-clock-parents = <&clk IMX8MQ_VIDEO_PLL1_OUT>,
-				 <&clk IMX8MQ_CLK_25M>;
-	assigned-clock-rate = <120000000>,
-			      <0>,
-			      <599999999>;
-
 	port@0 {
-		lcdif_mipi_dsi: mipi-dsi-endpoint {
+		lcdif_mipi_dsi: endpoint {
 			remote-endpoint = <&mipi_dsi_in>;
 		};
 	};
@@ -62,17 +53,6 @@
 	status = "okay";
 	as_bridge;
 	sync-pol = <1>;
-	assigned-clocks = <&clk IMX8MQ_CLK_DSI_PHY_REF>,
-			  <&clk IMX8MQ_CLK_DSI_CORE>,
-			  <&clk IMX8MQ_VIDEO_PLL1_REF_SEL>,
-			  <&clk IMX8MQ_VIDEO_PLL1>;
-	assigned-clock-parents = <&clk IMX8MQ_VIDEO_PLL1_OUT>,
-				 <&clk IMX8MQ_SYS1_PLL_266M>,
-				 <&clk IMX8MQ_CLK_25M>;
-	assigned-clock-rates = <24000000>,
-			       <266000000>,
-			       <0>,
-			       <599999999>;
 
 	port@1 {
 		mipi_dsi_in: endpoint {
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index 00f795c..0fc4277 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -551,12 +551,11 @@
 		clocks = <&clk IMX8MQ_CLK_DISP_APB_ROOT>,
 			 <&clk IMX8MQ_CLK_DISP_AXI_ROOT>,
 			 <&clk IMX8MQ_CLK_DISP_RTRM_ROOT>,
-			 <&clk IMX8MQ_CLK_DUMMY>,
 			 <&clk IMX8MQ_VIDEO2_PLL_OUT>,
 			 <&clk IMX8MQ_CLK_DISP_DTRC>,
 			 <&clk IMX8MQ_VIDEO2_PLL1_REF_SEL>,
 			 <&clk IMX8MQ_CLK_PHY_27MHZ>;
-		clock-names = "apb", "axi", "rtrm", "pix_div", "pix_out",
+		clock-names = "apb", "axi", "rtrm", "pix",
 			"dtrc", "pll_src", "pll_phy_ref";
 		assigned-clocks = <&clk IMX8MQ_CLK_DC_PIXEL>,
 				  <&clk IMX8MQ_CLK_DISP_AXI>,
@@ -635,8 +634,9 @@
 		interrupts = <GIC_SPI 34 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&clk IMX8MQ_CLK_DSI_PHY_REF>,
 			 <&clk IMX8MQ_CLK_DSI_AHB>,
-			 <&clk IMX8MQ_CLK_DSI_IPG_DIV>;
-		clock-names = "phy_ref", "rx_esc", "tx_esc";
+			 <&clk IMX8MQ_CLK_DSI_IPG_DIV>,
+			 <&clk IMX8MQ_VIDEO_PLL1>;
+		clock-names = "phy_ref", "rx_esc", "tx_esc", "video_pll";
 		assigned-clocks = <&clk IMX8MQ_CLK_DSI_AHB>;
 		assigned-clock-parents = <&clk IMX8MQ_SYS1_PLL_80M>;
 		assigned-clock-rates = <80000000>;
-- 
2.7.4

